clang/lib/CConv/ProgramVar.cpp

```diff
@@ -42,7 +44,20 @@ FunctionParamScope *FunctionParamScope::getFunctionParamScope(
   return FnParmScopeMap[MapK];
 }
 
-FunctionScope *FunctionScope::getFunctionScope(std::string FnName, bool IsSt) {
+CtxFunctionArgScope *CtxFunctionArgScope::getCtxFunctionParamScope(
```

Comment:
It looks like you index the map using a tuple constructed with the same information as the object you store in the map which feels a  bit weird to me. There might be a better way to do this.

Reply:
I wanted a singleton pattern here. I changed by using `set` and a custom comparator.